libsolutil/LEB128.h

```diff
@@ -0,0 +1,59 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+#pragma once
+
+
+#include <libsolutil/Common.h>
+
+namespace solidity::util
+{
+
+inline bytes lebEncode(uint64_t _n)
+{
+	// Based on https://github.com/llvm/llvm-project/blob/master/llvm/include/llvm/Support/LEB128.h
+	bytes result;
+	do
+	{
+		uint8_t v = _n & static_cast<uint8_t>(0x7f);
+		_n >>= static_cast<uint8_t>(7);
+		if (_n != 0)
+			v |= static_cast<uint8_t>(0x80); // Mark this byte to show that more bytes will follow.
+		result.emplace_back(char(v));
```

Comment:
Do we need the cast here? isn't `bytes` a `vector<uint8_t>`?

Reply:
Thats true, also the other casts are not really needed.